### DISCUSSION TITLE
Maria DB in bootloop after fresh install

### DISCUSSION THREAD

**Discussion by lunatikk1984** [photoprism/photoprism#4661] [Q&A]
Hi all,

I tried to do a fresh install using portainer on my qnap NAS. I tried to follow all the instructions from this link https://docs.photoprism.app/getting-started/portainer/ but ultimately failed because the mariadb container is in a bootloop always crashing with this error:

![image](https://github.com/user-attachments/assets/3205a6bf-4d36-410f-89fc-e0ca6daa6341)

I used the default portainer config from the getting-started link and only adapted my lan config, the volume mapping and I bumped up the innodb buffer pool size to 1G to increase performance.

this is my config file:

`
## PhotoPrism Stack Configuration for Portainer
services:
  photoprism:
    image: photoprism/photoprism:latest
    ## Don't enable automatic restarts until PhotoPrism has been properly configured and tested!
    ## If the service gets stuck in a restart loop, this points to a memory, filesystem, network, or database issue:
    ## https://docs.photoprism.app/getting-started/troubleshooting/#fatal-server-errors
    # restart: unless-stopped
    stop_grace_period: 10s
    depends_on:
      - mariadb
    security_opt:
      - seccomp:unconfined
      - apparmor:unconfined
    ## Server port mapping in the format "Host:Container". To use a different port, change the host port on
    ## the left-hand side and keep the container port, e.g. "80:2342" (for HTTP) or "443:2342 (for HTTPS):
    ports:
      - "2342:2342"
    ## Changes to the config options in stack.env always require a restart to take effect:
    ## https://docs.photoprism.app/getting-started/config-options/
    env_file:
      - stack.env
    working_dir: "/photoprism" # do not change or remove
    ## Storage Folders: "~" is a shortcut for your home directory, "." for the current directory
    volumes:
      - "/Photoprism/originals:/photoprism/originals" # Originals mount path can be changed as needed (DO NOT REMOVE)
      - "/Photoprism/storage:/photoprism/storage"     # *Writable* storage folder for cache, database, and sidecar files (DO NOT REMOVE)
      - "/Photoprism/import:/photoprism/import"       # Import folder for external files (added)
    networks:
      photoprism_net:
        ipv4_address: 192.168.178.73

  mariadb:
    image: mariadb:11
    ## If MariaDB gets stuck in a restart loop, this points to a memory or filesystem issue:
    ## https://docs.photoprism.app/getting-started/troubleshooting/#fatal-server-errors
    restart: unless-stopped
    stop_grace_period: 5s
    security_opt: # see https://github.com/MariaDB/mariadb-docker/issues/434#issuecomment-1136151239
      - seccomp:unconfined
      - apparmor:unconfined
    command: --innodb-buffer-pool-size=1G --transaction-isolation=READ-COMMITTED --character-set-server=utf8mb4 --collation-server=utf8mb4_unicode_ci --max-connections=512 --innodb-rollback-on-timeout=OFF --innodb-lock-wait-timeout=120
    ## Never store database files on an unreliable device such as a USB flash drive, an SD card, or a shared network folder:
    volumes:
      - "./database:/var/lib/mysql" # DO NOT REMOVE
    env_file:
      - stack.env

networks:
  photoprism_net:
    driver: bridge
    ipam:
      config:
        - subnet: 192.168.178.0/24
`

I used the env.yaml from here: https://dl.photoprism.app/docker/portainer/stack.env and I didn't change anything here.
The containers are deployed, but the mariaDb container won't start because of the error.

What am I doing wrong?

Best regards,

L

**Comment by omerdduran** [comment]
Hello. Firstly, review the troubleshooting guide for fatal server errors:
https://docs.photoprism.app/getting-started/troubleshooting/#fatal-server-errors

Also pay attention to these:
- Never store database files on a network share due to possible caching/synchronisation issues
- Delete any partial database files before retrying
- Make sure the database directory has the appropriate permissions
- Based on past cases, removing the InnoDB log files (ib_logfile0 and ib_logfile1) from your database directory and allowing MariaDB to recreate them may solve the problem.

**Reply by lunatikk1984** [reply]
> Hello. Firstly, review the troubleshooting guide for fatal server errors: https://docs.photoprism.app/getting-started/troubleshooting/#fatal-server-errors
> 
> Also pay attention to these:
> 
> * Never store database files on a network share due to possible caching/synchronisation issues
> * Delete any partial database files before retrying
> * Make sure the database directory has the appropriate permissions
> * Based on past cases, removing the InnoDB log files (ib_logfile0 and ib_logfile1) from your database directory and allowing MariaDB to recreate them may solve the problem.

I know this troubleshooting section and it didn't help at all, unfortunately (I also couldn't find any of the errors from the log in the first post). Is there any detailed description to those extra steps you posted?

I'm not able to connect with a console to the container to perform any of those tasks, because it's constantly restarting.

I don't know what I'm missing, I thought this is going to be straightforward :/

Best regards,

L
